packages/core/src/2d/sprite/Sprite.ts

```diff
@@ -0,0 +1,232 @@
+import { Vector2, Vector4, Rect } from "@oasis-engine/math";
+import { RefObject } from "../../asset/RefObject";
+import { Engine } from "../../Engine";
+import { Texture2D } from "../../texture";
+
+export class Sprite extends RefObject {
+  private static _tempVec2: Vector2 = new Vector2();
+
+  private static _VERTICES_FLAG = 0x1;
```

Comment:
You can  write a internal enum in this  file.

Reply:
> You can write a internal enum in this file.

done